### PR TITLE
Update checkpoint_converter_fsdp_hf.py

### PR DESCRIPTION
**Pull request by Mattral**
# What does this PR do?

This PR refactors the existing script to improve modularity, readability, and error handling.
[src/llama_recipes/inference/checkpoint_converter_fsdp_hf.py]

### Description of the Change
- **Modularization**: Introduced `get_model_name_from_yaml` and `load_and_save_model` functions to enhance code structure.
- **Exception Handling**: Added specific exception handling for improved error reporting.
- **Code Readability**: Grouped imports and added docstrings for better understanding.
- **Logging**: Included print statements for logging key steps (consider future use of the `logging` module for more control).

### Motivation and Context
These changes improve the readability, maintainability, and robustness of the code, making it easier to understand and extend. The refactored code also handles errors more gracefully, providing clearer messages to the user.

### Dependencies
No new dependencies are required for this change.

# Summary of Changes

## Key Improvements

1. **Modularization**:
   - Introduced `get_model_name_from_yaml` and `load_and_save_model` functions to improve code modularity and readability.

2. **Exception Handling**:
   - Enhanced exception handling to be more specific and informative.

3. **Code Readability**:
   - Grouped similar imports together and added docstrings for better understanding.

4. **Logging**:
   - Included print statements for logging key steps (consider future use of the `logging` module for more control).

### Import Optimizations
- Grouped standard library, third-party, and local application imports.
- Removed unnecessary commented-out imports.

### Function Decomposition
- Split the main logic into smaller functions: `get_model_name_from_yaml` and `load_and_save_model`.

### Exception Handling
- Improved error handling in `get_model_name_from_yaml` to handle specific exceptions like `FileNotFoundError` and general exceptions.

### Docstrings and Comments
- Added docstrings to functions to describe their purpose and usage.
- Improved inline comments for clarity.

### CODE REVIEW

File: src/llama_recipes/inference/checkpoint_converter_fsdp_hf.py [hunk 1]
```diff
@@ -3,14 +3,16 @@
 
 # from accelerate import init_empty_weights, load_checkpoint_and_dispatch
 
-import fire
 import os
 import sys
 import yaml
+import fire
```
Comment (mreso) [inline src/llama_recipes/inference/checkpoint_converter_fsdp_hf.py:9]: The imports where previously sorted alphabetically. Any reason to change this?

File: src/llama_recipes/inference/checkpoint_converter_fsdp_hf.py [hunk 2]
```diff
@@ -3,14 +3,16 @@
...
 
 from transformers import AutoTokenizer
 
-from llama_recipes.inference.model_utils import  load_llama_from_config
+from llama_recipes.inference.model_utils import load_llama_from_config
+
+from model_checkpointing import load_sharded_model_single_gpu
```
Comment (mreso) [inline src/llama_recipes/inference/checkpoint_converter_fsdp_hf.py:15]: This import will fail as the path is not added sys.path in line 22/24 (which is something we should get rid of in any case). Lets just change it to from llama_recipes.model_checkpointing import load_sharded_model_single_gpu

**Review by mreso — CHANGES_REQUESTED** [review]
Thanks for the contribution. Love to continue reviewing this but can you please add some logs first to the PR showing that it runs? This will show if there are any further hidden errors like the one with the "from model_checkpointing" import I commented on.

**Comment by Karliz24** [comment]
El día

Obtener Outlook para iOS<https://aka.ms/o0ukef>
________________________________
De: Facebook Community Bot ***@***.***>
Enviado: Monday, August 5, 2024 11:53:03 PM
Para: meta-llama/llama-recipes ***@***.***>
Cc: Subscribed ***@***.***>
Asunto: Re: [meta-llama/llama-recipes] Update checkpoint_converter_fsdp_hf.py (PR #622)


Hi @Mattral<https://github.com/Mattral>!

Thank you for your pull request and welcome to our community.

Action Required

In order to merge any pull request (code, docs, etc.), we require contributors to sign our Contributor License Agreement, and we don't seem to have one on file for you.

Process

In order for us to review and merge your suggested changes, please sign at https://code.facebook.com/cla. If you are contributing on behalf of someone else (eg your employer), the individual CLA may not be sufficient and your employer may need to sign the corporate CLA.

Once the CLA is signed, our tooling will perform checks and validations. Afterwards, the pull request will be tagged with CLA signed. The tagging process may take up to 1 hour after signing. Please give it that time before contacting us about it.

If you have received this in error or have any questions, please contact us at ***@***.******@***.***?subject=CLA%20for%20meta-llama%2Fllama-recipes%20%23622>. Thanks!

**Comment by Karliz24** [comment]
Obtener Outlook para iOS<https://aka.ms/o0ukef>
________________________________
De: Min Htet Myet (Mattral) ***@***.***>
Enviado: Monday, August 5, 2024 11:52:55 PM
Para: meta-llama/llama-recipes ***@***.***>
Cc: Subscribed ***@***.***>
Asunto: [meta-llama/llama-recipes] Update checkpoint_converter_fsdp_hf.py (PR #622)

What does this PR do?

This PR refactors the existing script to improve modularity, readability, and error handling.
[src/llama_recipes/inference/checkpoint_converter_fsdp_hf.py]

Description of the Change

  *   Modularization: Introduced get_model_name_from_yaml and load_and_save_model functions to enhance code structure.
  *   Exception Handling: Added specific exception handling for improved error reporting.
  *   Code Readability: Grouped imports and added docstrings for better understanding.
  *   Logging: Included print statements for logging key steps (consider future use of the logging module for more control).

Motivation and Context

These changes improve the readability, maintainability, and robustness of the code, making it easier to understand and extend. The refactored code also handles errors more gracefully, providing clearer messages to the user.

Dependencies

No new dependencies are required for this change.

Summary of Changes
Key Improvements

  1.  Modularization:

     *   Introduced get_model_name_from_yaml and load_and_save_model functions to improve code modularity and readability.
  2.  Exception Handling:

     *   Enhanced exception handling to be more specific and informative.
  3.  Code Readability:

     *   Grouped similar imports together and added docstrings for better understanding.
  4.  Logging:

     *   Included print statements for logging key steps (consider future use of the logging module for more control).

Detailed Changes
Import Optimizations

  *   Grouped standard library, third-party, and local application imports.
  *   Removed unnecessary commented-out imports.

Function Decomposition

  *   Split the main logic into smaller functions: get_model_name_from_yaml and load_and_save_model.

Exception Handling

  *   Improved error handling in get_model_name_from_yaml to handle specific exceptions like FileNotFoundError and general exceptions.

Docstrings and Comments

  *   Added docstrings to functions to describe their purpose and usage.
  *   Improved inline comments for clarity.

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/meta-llama/llama-recipes/pull/622

Commit Summary

  *   c4bfd4b<https://github.com/meta-llama/llama-recipes/pull/622/commits/c4bfd4bee7c9bdf847a6da307cc9e3cb116596ea> Update checkpoint_converter_fsdp_hf.py

File Changes

(1 file<https://github.com/meta-llama/llama-recipes/pull/622/files>)

  *   M src/llama_recipes/inference/checkpoint_converter_fsdp_hf.py<https://github.com/meta-llama/llama-recipes/pull/622/files#diff-a41b04789be2c6a96adb9fa76fb02ddfd845a93208afc32d086536143aa88752> (90)

Patch Links:

  *   https://github.com/meta-llama/llama-recipes/pull/622.patch
  *   https://github.com/meta-llama/llama-recipes/pull/622.diff

**Comment by init27** [comment]
@Mattral please let us know if you have the chance to update per the comments above-thanks!

**Comment by Mattral** [comment]
I will update those following day,
sorry for delay.

**Pull request closed without merging**